### PR TITLE
[WIP] Added support for input types parsing and code generation (#186)

### PR DESCRIPTION
**Pull request by atapin**

### CODE REVIEW

File: core/src/main/scala/caliban/parsing/Parser.scala [hunk 1]
```diff
@@ -192,7 +192,7 @@ object Parser {
     }
 
   private def typeDefinition[_: P]: P[TypeDefinition] =
-    P("type" ~/ name ~ "{" ~ fieldDefinition.rep ~ "}").map(t => TypeDefinition(t._1, t._2.toList))
+    P(("type" | "input") ~/ name ~ "{" ~ fieldDefinition.rep ~ "}").map(t => TypeDefinition(t._1, t._2.toList))
```
Comment (ghostdogpr) [inline core/src/main/scala/caliban/parsing/Parser.scala]: Actually if you look at the spec, an input type is slightly different from a regular type:
http://spec.graphql.org/June2018/#InputValueDefinition
http://spec.graphql.org/June2018/#FieldDefinition
(no arguments possible, default value added)

**Comment by ghostdogpr** [comment]
@atapin Hi! I did all of the missing types (enum, union, scalar, etc) so this is the only one left. Do you mind if I do it while I'm at it?

**Comment by atapin** [comment]
@ghostdogpr sure. I was gonna ask for help with adding the types

**Comment by ghostdogpr** [comment]
Thanks! I'll add it soon in #209 
Closing this one then.

**Pull request closed without merging**